src/app/ad-authoring/ad-authoring.state.ts

```diff
@@ -9,15 +9,177 @@ export interface AdAuthoringWorkflowState {
   readonly callToAction?: CallToActionEnum;
   readonly fileSrc?: string;
   readonly file?: File;
+  readonly AMPHTMLstring: string;
 }
-// add a default state for the fields above
+
 @Injectable({
   providedIn: 'root',
 })
 export class AdAuthoringWorkflowStateContainer {
+  generateAMPHTML(
+    callToActionStr: string,
+    landingPageUrl: string,
+    landingPageType: string,
+    base64AssetStr: string,
+    assetFile: File
+  ) {
+    let adAMPHTML =
+      '"<!doctype html><html amp4ads><head><meta charset=\\"utf-8\\"><meta name=\\"viewport\\" content=\\"width=device-width,minimum-scale=1\\"><meta name=\\"amp-cta-type\\" content=\\"' +
+      callToActionStr +
+      '\\"><meta name=\\"amp-cta-url\\" content=\\"' +
+      landingPageUrl +
+      '\\"><meta name=\\"amp-cta-landing-page-type\\" content=\\"' +
+      landingPageType +
+      '\\"><style amp4ads-boilerplate>body{visibility:hidden}<\\/style><script async src=\\"https:\\/\\/cdn.ampproject.org\\/amp4ads-v0.js\\"><\\/script><\\/head><body><p>Hello, fake ad with srcdoc<\\/p><amp-img layout=\\"fixed\\" height=\\"250\\" width=\\"300\\" src=\\"' +
+      base64AssetStr +
+      '\\"><\\/amp-img><\\/body><\\/html>"';
+
+    // if the file uploaded is a video (not image)
+    if (assetFile != null && assetFile.type.includes('video')) {
```

Comment:
We can restructure this to reduce repetition. The only difference between the first adAMPHTML assignment and the second one is the asset info. We can calculate the asset HTML that should be appended to the rest of the ad HTML. 

Something like:
const assetHtml = // if assetFile & isVideo ? videoHtml : imageHtml
const adAmpHtml = // adAmpHtml + assetHtml